### PR TITLE
update elasticsearch python client to 8.16

### PR DESCRIPTION
**Pull request by miguelgrinberg**
Part of the 8.16.0 release tasks. See https://github.com/elastic/search-team/issues/8045

#### Pre-Review Checklist
- [ ] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [ ] this PR has a meaningful title
- [ ] this PR links to all relevant github issues that it fixes or partially addresses
- [ ] if there is no GH issue, please create it. Each PR should have a link to an issue
- [ ] this PR has a thorough description
- [ ] Covered the changes with automated tests
- [ ] Tested the changes locally
- [ ] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference
- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)

#### Changes Requiring Extra Attention

- [ ] Security-related changes (encryption, TLS, SSRF, etc)
- [ ] New external service dependencies added.

### CODE REVIEW

**Comment by seanstory** [comment]
Well this is fun. Tests are failing with:
```
FAILED tests/test_kibana.py::test_main - elasticsearch.exceptions.GeneralAvailabilityWarning: This API is in beta and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.
--
```

@pquentin this PR just bumps python es client versions, doesn't change any usage. Why's it now raising that we're hitting beta APIs, if it wasn't before?

**Comment by artem-shelkovnikov** [comment]
cc @jedrazb is it something coming from connectors API usage?

**Comment by miguelgrinberg** [comment]
@seanstory This is correct, the 8.16 Python client is the first to warn about usage of non-GA endpoints.

**Review by pquentin — APPROVED** [review]

**Comment by pquentin** [comment]
Sorry I approved before seeing messages or CI. I can show you how to ignore the warning in pytest if you are interested.

**Comment by artem-shelkovnikov** [comment]
I will go ahead and ignore the test :)

**Comment by artem-shelkovnikov** [comment]
@pquentin will this mean that our connector logs will log warnings during the calls? if so, do we need to pass some options to ignore the warning for our main app?

**Comment by pquentin** [comment]
The connectors API are still in beta, but it makes sense for you to use them, right. In that case, an application-wide ignore could be the best option, as documented in the release notes:

```python
import warnings
from elasticsearch.exceptions import GeneralAvailabilityWarning

warnings.filterwarnings("ignore", category=GeneralAvailabilityWarning)
```

**Review by artem-shelkovnikov — APPROVED** [review]

File: requirements/framework.txt [hunk 1]
```diff
@@ -3,8 +3,8 @@ aiofiles==23.2.1
 aiomysql==0.1.1
 httpx==0.27.0
 httpx-ntlm==1.4.0
-elasticsearch[async]==8.15.0
-elastic-transport==8.15.0
+elasticsearch[async]==8.16.0
+elastic-transport==8.15.1
```
Comment (jedrazb) [inline requirements/framework.txt:7]: I might have context missing, but shouldn't this be 8.16.0 as well?

**Review by jedrazb — APPROVED** [review]

**Pull request merged**